src/components/LayoutTree.tsx

```diff
@@ -567,12 +580,22 @@ const LayoutTree = () => {
         ycol1Ancestors.push({ y: yaccum + height, depth: node.depth })
       }
 
+      // check if the previous thought is already the last visible thought
+      const prevIsVisible = isVisible(prev?.autofocus, node.autofocus)
+
+      // check if the next thought is visible or not
+      const nextIsLastVisible = isVisible(next?.autofocus, node.autofocus)
+
+      // if previous thought is last visible then set isLastVisible of every next thought to false
+      const isLastVisible = !prevIsVisible ? false : !nextIsLastVisible
```

Comment:
I think this can be simplified to the following, or am I missing a case?

```ts
const isLastVisible =
  (node.autofocus === 'dim' || node.autofocus === 'show') &&
  !(next?.autofocus === 'dim' || next?.autofocus === 'show')
```

Reply:
Yes, it works great. Thanks